4.sc_Morphology_Neural_Network_MLP_Model/0.hyperparameter_optimization/scripts/Hyperparameter_Optimization_model_regression.py

```diff
@@ -0,0 +1,341 @@
+# ### Here I will be predicting if a cell received a treatment or not
+
+# %%
+import pathlib
+import sys
+
+import numpy as np
+import optuna
+import pandas as pd
+import pyarrow.parquet as pq
+import toml
+import torch
+from sklearn import preprocessing
+
+sys.path.append("../..")
+
+from MLP_utils.parameters import Parameters
+from MLP_utils.utils import (
+    Dataset_formatter,
+    data_split,
+    extract_best_trial_params,
+    objective_model_optimizer,
+    parameter_set,
+    plot_metric_vs_epoch,
+    results_output,
+    test_optimized_model,
+    train_optimized_model,
+    un_nest,
+)
+
+sys.path.append("../../..")
+from utils.utils import df_stats
+
+# %% [markdown]
+# #### Set up Data to be compatible with model
+
+# %% [markdown]
+# ##### Regression Model Data Wrangling and Set Up
+# comment out if not using regression
+
+# %%
+# Parameters
+CELL_TYPE = "PBMC"
+CONTROL_NAME = "DMSO_0.100_DMSO_0.025"
+# TREATMENT_NAME = "LPS_100.000_DMSO_0.025"
+TREATMENT_NAME = "LPS_100.000_DMSO_0.025"
+MODEL_NAME = "DMSO_0.025_vs_LPS_100"
+
+# %%
+ml_configs_file = pathlib.Path("../../MLP_utils/regression_config.toml").resolve(
+    strict=True
+)
+ml_configs = toml.load(ml_configs_file)
+params = Parameters()
+mlp_params = parameter_set(params, ml_configs)
+
+# overwrite params via command line arguments from papermill
+mlp_params.CELL_TYPE = CELL_TYPE
+mlp_params.MODEL_NAME = MODEL_NAME
+mlp_params.CONTROL_NAME = CONTROL_NAME
+mlp_params.TREATMENT_NAME = TREATMENT_NAME
+mlp_params.SHUFFLE = False
+
+# %%
+# Import Data
+# set data file path under pathlib path for multi-system use
+
+# Commented out for now, using a different data set to trobleshoot
+# file_path = pathlib.Path(
+#     f"../../../data/{mlp_params.CELL_TYPE}_preprocessed_sc_norm.parquet"
+# ).resolve(strict=True)
+
+file_path = pathlib.Path(
+    "../../../data/PBMC_subset_sc_norm_DMSO_0.100_DMSO_0.025_LPS_100.000_DMSO_0.025.parquet"
+)
+
+# set path for nomic data
+nomic_df_path = pathlib.Path(
+    f"../../../2.Nomic_nELISA_Analysis/Data/clean/Plate2/nELISA_plate_430420_{mlp_params.CELL_TYPE}_cleanup4correlation.csv"
+).resolve(strict=True)
+
+df = pq.read_table(file_path).to_pandas()
+nomic_df = pd.read_csv(nomic_df_path)
+
+# %%
+# change the nomic df to standard scaler
+# select the columns that contain "NSU"
+nomic_df_scaled = nomic_df.filter(regex="NSU")
+# standardize the nomic data
+# scaler = preprocessing.StandardScaler()
+# nomic_df_scaled = pd.DataFrame(scaler.fit_transform(nomic_df_scaled), columns=nomic_df_scaled.columns)
+# add the nomic data metadata back
+nomic_df_scaled[
+    [
+        "Metadata_position_x",
+        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose",
+        "fourb_Metadata_Treatment_Dose_Inhibitor_Dose",
+    ]
+] = nomic_df[
+    [
+        "Metadata_position_x",
+        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose",
+        "fourb_Metadata_Treatment_Dose_Inhibitor_Dose",
+    ]
+]
+nomic_df = nomic_df_scaled.copy()
+del nomic_df_scaled
```

Comment:
Seeing the method of copying and deleting data here made me wonder if another approach could help. Consider the use of a list comprehension over the column names or `pd.DataFrame.filter()` using regex to create the df.